packages/react-ui-core/src/Ratings/RatingBar.js

```diff
@@ -2,60 +2,54 @@ import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 import themed from 'react-themed'
 import classnames from 'classnames'
-import Star from './Star'
 
 @themed(/^(RatingBar|Label)/, {
   pure: true,
 })
 
 export default class RatingBar extends PureComponent {
   static propTypes = {
-    uniqueId: PropTypes.string.isRequired,
     theme: PropTypes.object,
-    fillColor: PropTypes.string,
     className: PropTypes.string,
     maxScore: PropTypes.number,
     score: PropTypes.number,
     label: PropTypes.string,
-    RatingItem: PropTypes.func,
   }
   static defaultProps = {
     theme: {},
     score: 0,
     maxScore: 5,
-    RatingItem: Star,
-    fillColor: 'black',
   }
 
   get ratingItems() {
     const {
-      uniqueId,
-      maxScore,
-      RatingItem,
       theme,
-      className,
+      score,
+      maxScore,
       ...props
     } = this.props
 
-    return [...Array(maxScore).keys()].map(index => (
-      <RatingItem
-        key={index}
-        className={
-          classnames(
-            theme.RatingBar_Item,
-            theme[`RatingBar_Item-${index}`],
-            this.props.fillColor && theme[`RatingBar_Item-${this.props.fillColor}`]
-          )
-        }
-        uniqueId={`${uniqueId}-rating-item-${index}`}
-        width={`${this.fillWidth(index)}%`}
-        {...props}
-      />
-    ))
+    const calcScore = (score / maxScore)
+    const scorePercent = (calcScore > 1 ? 1 : calcScore) * 100
+
+    return (
+      <div
+        className={theme.RatingBar_Background} {...props}
```

Comment:
I think you need to add `className` back to the const above, so it is not included in `props`, otherwise it overrides the `className` you are setting.

Reply:
And on second thought, it seems a little strange to spread `{...props}` onto an interior element rather than onto the top-most element of the component. I supposed this is being done because you need the `onClick` to be attached here to take advantage of the positioning?

Reply:
As an alternative, see #275. I think it doesn't add much extra code and moves the calculation of which star was clicked into this component. But I'm okay with either approach.

Reply:
@barelyfitz you're correct about the `className`. It really should be added to the root node.  Also, by spreading `{...props}` it would override everything before it including `className`. So yeah, things you don't want to overwrite should be _after_ `{...props}`.